kodein-core/src/main/kotlin/com/github/salomonbrys/kodein/types.kt

```diff
@@ -69,16 +69,21 @@ internal abstract class JVMTypeToken<T> : TypeToken<T> {
 
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
+        if (other is CompositeTypeToken<*>) {
+            return fullDispString().equals(this.fullDispString())
```

Comment:
compare between general type token and composite type tonken with their's fullDispString.
generic<A<B>>() == erasedComp1<A<B>>(erased<A>(), erased<B>())

Reply:
This is very interesting !